src/transformers/models/sam/convert_sam_original_to_hf_format.py

```diff
@@ -88,73 +133,71 @@ def replace_keys(state_dict):
-def convert_sam_checkpoint(model_name, pytorch_dump_folder, push_to_hub, model_hub_id="ybelkada/segment-anything"):
-    checkpoint_path = hf_hub_download(model_hub_id, f"checkpoints/{model_name}.pth")
-
-    if "sam_vit_b" in model_name:
-        config = SamConfig()
-    elif "sam_vit_l" in model_name:
-        vision_config = SamVisionConfig(
-            hidden_size=1024,
-            num_hidden_layers=24,
-            num_attention_heads=16,
-            global_attn_indexes=[5, 11, 17, 23],
-        )
-
-        config = SamConfig(
-            vision_config=vision_config,
-        )
-    elif "sam_vit_h" in model_name:
-        vision_config = SamVisionConfig(
-            hidden_size=1280,
-            num_hidden_layers=32,
-            num_attention_heads=16,
-            global_attn_indexes=[7, 15, 23, 31],
-        )
+def convert_sam_checkpoint(model_name, pytorch_dump_folder, push_to_hub):
+    if model_name == "slimsam-50-uniform":
+        checkpoint_path = "/Users/nielsrogge/Documents/SlimSAM/SlimSAM-50-uniform.pth"
+    elif model_name == "slimsam-77-uniform":
+        checkpoint_path = "/Users/nielsrogge/Documents/SlimSAM/SlimSAM-77-uniform.pth"
+    else:
+        checkpoint_path = hf_hub_download("ybelkada/segment-anything", f"checkpoints/{model_name}.pth")
 
-        config = SamConfig(
-            vision_config=vision_config,
-        )
+    config = get_config(model_name)
 
     state_dict = torch.load(checkpoint_path, map_location="cpu")
     state_dict = replace_keys(state_dict)
 
     image_processor = SamImageProcessor()
-
     processor = SamProcessor(image_processor=image_processor)
     hf_model = SamModel(config)
+    hf_model.eval()
+
+    device = "cuda" if torch.cuda.is_available() else "cpu"
 
     hf_model.load_state_dict(state_dict)
-    hf_model = hf_model.to("cuda")
+    hf_model = hf_model.to(device)
 
     img_url = "https://huggingface.co/ybelkada/segment-anything/resolve/main/assets/car.png"
     raw_image = Image.open(requests.get(img_url, stream=True).raw).convert("RGB")
 
-    input_points = [[[400, 650]]]
+    input_points = [[[500, 375]]]
     input_labels = [[1]]
 
-    inputs = processor(images=np.array(raw_image), return_tensors="pt").to("cuda")
+    inputs = processor(images=np.array(raw_image), return_tensors="pt").to(device)
 
     with torch.no_grad():
         output = hf_model(**inputs)
     scores = output.iou_scores.squeeze()
 
-    if model_name == "sam_vit_h_4b8939":
-        assert scores[-1].item() == 0.579890251159668
+    if model_name == "sam_vit_b_01ec64":
+        inputs = processor(
+            images=np.array(raw_image), input_points=input_points, input_labels=input_labels, return_tensors="pt"
+        ).to(device)
+
+        with torch.no_grad():
+            output = hf_model(**inputs)
+            scores = output.iou_scores.squeeze()
+
+        print(scores)
+        pass
+
+    elif model_name == "sam_vit_h_4b8939":
+        # assert scores[-1].item() == 0.579890251159668
```

Comment:
```suggestion
```

Reply:
Hi I think it's a bad practice to remove assertion checks from conversion scripts.

e.g. for llava 1.5 or SAM, there's no proof that we match the original logits (and I wasn't also able to match them - for llava 1.5 it turned out to be because of the image which wasn't padded as done in the original repo)

Hence I'm advocating to keep assertions in conversion scripts (as they are mostly run by us)